skyportal/handlers/api/plan.py

```diff
@@ -0,0 +1,450 @@
+
+    params["doMinimalTiling"] = True
+    params["doIterativeTiling"] = False
+    params["galaxies_FoV_sep"] = 1.0
+    params["doMaxTiles"] = doMaxTiles
+    params["max_nb_tiles"] = np.array([max_nb_tiles] * len(filt))
+
+    if params["doEvent"]:
+        params["skymap"], eventinfo = gwemopt.gracedb.get_event(params)
+        params["gpstime"] = eventinfo["gpstime"]
+        event_time = time.Time(params["gpstime"], format='gps', scale='utc')
+        params["dateobs"] = event_time.iso
+    elif params["doSkymap"]:
+        event_time = time.Time(params["gpstime"], format='gps', scale='utc')
+        params["dateobs"] = event_time.iso
+    elif params["doFootprint"]:
+        params["skymap"] = gwemopt.footprint.get_skymap(params)
+        event_time = time.Time(params["gpstime"], format='gps', scale='utc')
+        params["dateobs"] = event_time.iso
+    elif params["doDatabase"]:
+        event_time = time.Time(params["dateobs"], format='datetime', scale='utc')
+        params["gpstime"] = event_time.gps
+    else:
+        raise ValueError(
+            'Need to enable --doEvent, --doFootprint, ' '--doSkymap, or --doDatabase'
+        )
+
+    if tobs is None:
+        now_time = time.Time.now()
+        timediff = now_time.gps - event_time.gps
+        timediff_days = timediff / 86400.0
+        params["Tobs"] = np.array([timediff_days, timediff_days + 1])
+    else:
+        params["Tobs"] = tobs
+
+    params["doSingleExposure"] = True
+    if filterScheduleType == "block":
+        params["doAlternatingFilters"] = True
+    else:
+        params["doAlternatingFilters"] = False
+    params["filters"] = filt
+    params["exposuretimes"] = exposuretimes
+    params["mindiff"] = mindiff
+
+    params["doRASlice"] = doRASlice
+    params["raslice"] = raslice
+
+    params = gwemopt.segments.get_telescope_segments(params)
+    params = gwemopt.utils.params_checker(params)
+
+    if params["doPlots"]:
+        if not os.path.isdir(params["outputDir"]):
+            os.makedirs(params["outputDir"])
+
+    return params
+
+
+def gen_structs(params):
+
+    print('Loading skymap')
+    # Function to read maps
+    map_struct = gwemopt.utils.read_skymap(
+        params, is3D=params["do3D"], map_struct=params['map_struct']
+    )
+    print('Generating MOC struct')
+    moc_structs = gwemopt.skyportal.create_moc_from_skyportal(
+        params, map_struct=map_struct
+    )
+    tile_structs = gwemopt.tiles.moc(params, map_struct, moc_structs)
+
+    if "previous_coverage_struct" in params:
+        tile_structs, coverage_struct = gwemopt.coverage.timeallocation(
+            params,
+            map_struct,
+            tile_structs,
+            previous_coverage_struct=params["previous_coverage_struct"],
+        )
+    else:
+        tile_structs, coverage_struct = gwemopt.coverage.timeallocation(
+            params, map_struct, tile_structs
+        )
+
+    if params["doPlots"]:
+        gwemopt.plotting.skymap(params, map_struct)
```

Comment:
so these calls are modifying the arguments?

Reply:
This was dumb. I am removing.